x/staking/keeper/grpc_query.go

```diff
@@ -505,7 +505,7 @@ func queryRedelegation(ctx context.Context, k Querier, req *types.QueryRedelegat
 		return nil, err
 	}
 
-	redel, err := k.GetRedelegation(ctx, delAddr, srcValAddr, dstValAddr)
+	redel, err := k.Keeper.Redelegations.Get(ctx, collections.Join3(sdk.AccAddress(delAddr), sdk.ValAddress(srcValAddr), sdk.ValAddress(dstValAddr)))
```

Comment:
random q, are we able to make the keys only be bytes instead of sdk.AccAddress and sdk.ValAddress? its just a type alias so dont see it being a problem

Reply:
I'm not sure about it, but we might need to introduce another helper method to append length of address because of legacy key construction should also work.

cc: @testinginprod

Reply:
we should be able to, yes: https://github.com/cosmos/cosmos-sdk/blob/main/types/collections.go#L52